### ISSUE TITLE
ToU mobile rollout pre-validation

### ISSUE THREAD

**Issue by TKDickson** [department-of-veterans-affairs/va-mobile-app#8665]
A lot of the testing has already been done (several months ago); when the identity team reached out to talk about end of May 2024 rollout to the mobile app, we uncovered a case that needs checking:

> what Trevor said above: "one caveat when the rollout happens, users won't be able to create or refresh their session without accepting terms. So within 30 minutes of ToU being enabled everyone using the mobile app will be forced to re-authenticate and accept terms to continue using it"
> Has session expiration and re-authentication in the mobile app been tested, either specifically for this rollout or in general?  I'd like to know if we're confident that this will work seamlessly when this pops up for veterans.

My test plan
Android
- [ ] User who's actively using the app (foreground session) when ToU flag flipped
            Continue using the app/making API calls. Either I should be able to successfully use the app for 30 minutes, then gracefully need to re-auth; or at the next token refresh from the mobile side/next API call, I may be forced to re-auth.
- [ ] User with an active session no longer using the app (background session) when ToU flag flipped
             Wait 31 minutes, then bring the app to the foreground. I anticipate it'll be logged out and I'll need to re-auth.

iOS
- [ ] User who's actively using the app (foreground session) when ToU flag flipped
            Continue using the app/making API calls. Either I should be able to successfully use the app for 30 minutes, then gracefully need to re-auth; or at the next token refresh from the mobile side/next API call, I may be forced to re-auth.
- [ ] User with an active session no longer using the app (background session) when ToU flag flipped
             Wait 31 minutes, then bring the app to the foreground. I anticipate it'll be logged out and I'll need to re-auth.

I will have biometrics enabled for all of these users. I assume that biometrics will NOT work, and they'll be forced to UN/PW to see ToU, but that's just a hunch/I want to actually see the results.

**Comment by TKDickson** [comment]
Have a half hour blocked with someone from the identity team on Monday May 20. I'll need to send Trevor Bosaw either the uuids or ICNs of my users, so he can mark them as declined on the backend (mimics the relevant rollout workflow) during the designated window.

**Comment by TKDickson** [comment]
royal.tester (iPhone, foreground): 04b76edcebfe4b589ec0364c2351443f
+73 (iPad, background): bb708be81f9c4e0d9e4e9b772a9bfb7b
+7 (one plus, background): 1c1d6c19da6340fbb393ab687d27c22d
+119 (pixel, foreground): d45fb407976e47a78daa40944943ff0f

**Comment by TKDickson** [comment]
All users were logged out within the roughly 30 minute window (22 minutes and 28 minutes for the users I was actively maintaining a session with; the two users I checked after 30 minutes was up were logged out on their first API-calling action after that).

No users were allowed to use biometrics to log in. All of them successfully logged in, accepted terms of use (or for one user - declined, then logged in to accept, ToU), and were back in the mobile app.

Testing limitation - I only tried with SIS + id.me (no IAM, and no other auth services because of lack of staging users with viable other methods for mobile).

Closing this ticket as complete, and will report back in the DSVA slack thread.
